### ISSUE TITLE
Messages shouldn't be suppressed in programmatic changes

### ISSUE THREAD

**Issue by rodrigogiraoserrao** [Textualize/textual#2500]
TL;DR: We should **not** prevent a message from being posted just because the state of the widget was changed _programmatically_, and _even though_ the programmer already knows the state changed.

Immediate effect: `TabbedContent` will now post `TabActivated` when the reactive attribute `active` is assigned to.

Possible further considerations:
 - Add a message `Completed` to the `ProgressBar` widget.
 - Add a message `Started` to the `ProgressBar` widget.
 - ...?

Thesis:
Most widgets will post appropriate messages when their state changes, even if through programmatic means.
E.g.,
 - `DataTable` will post `CellHighlighted` if we move the cursor with `.move_cursor` or by assigning directly to `cursor_coordinate`;
 - `Checkbox` and `RadioButton` will post `Changed` if we toggle them with `.toggle` or by assigning directly to `value`; and
 - `OptionList` will post `OptionHighlighted` if we assign to the reactive `highlighted`.

I think all programmatic changes should still post the appropriate messages that reflect the changes that happened.

Even if a programmer does changes like this programmatically, they are sensible in assuming that the respective events will be triggered.
This lets them decouple two things:
 - the logic that _will change_ some state; and
 - the logic that _handles_ changes to the state.

If a programmer does _not_ want the message, they can explicitly opt-out with the context manager `prevents`, which is why the context manager was created in the first place.

<details>
<summary>To consider if this is accepted.</summary>

 1. For the issue with `TabbedContent`: the usage of `prevent` in the `TabbedContent` comes from #2305 and #2322, so take those two PRs into account when fixing this.
 2. Consider if there are other events missing that a user might want to be notified about.

</details>

**Comment by rodrigogiraoserrao** [comment]
Settled with #3417.

We post all messages.
